DirectEmployers/MyJobs
# [PD 3168] implement customerid control

## What's this PR do?

Destroys everything you love.

Just kidding. But it does make using CBO a little less convenient.

This PR adds CustomerID handling/enforcement to the CBO module. Rather than using a hardcoded CustomerID to display CBOs and such, the CustomerID will now be mapped to the currently selected company (via Toolbar/user status).

## Screenshots (if appropriate)

## Where should the reviewer start?
NOTE: This step may no longer be necessary. If there are any issues with accessing the CBO module with your currently selected company, follow these steps.

Start by running this script in your Django shell. Be sure to put your email in the proper place.

```
from myjobs.models import AppAccess, User
from seo.models import Company

de = Company.objects.get(name='AAR Corporation')
aa = AppAccess.objects.get(name='cbo') # Or any other appaccess
de.app_access.add(aa)

usr = User.objects.get(email="ENTER YOUR EMAIL")
dea = de.role_set.objects.all()[0] # grabs admin role
usr.roles.add(dea)
```

Once this is done, go to secure.my.jobs, select AAR as your company and access CBO.

## How should this be manually tested?

Well.. See above. You can also try selecting a company that doesn't have a CustomerID yet (like DirectEmployers) and see the pretty error it gives.

## What's Next

This PR also sets the stage for the opt in/opt out controls for CBO. This PR does not rely on that control, however, and can be merged without it.
